cardano-node/src/Cardano/Tracing/Tracers.hs

```diff
@@ -413,18 +446,27 @@ mkTracers traceConf tracer txsProcessedCounter = do
         meta <- mkLOMeta Critical Confidential
         traceNamedObject tr' (meta, logValue1)
         traceNamedObject tr' (meta, logValue2)
- 
-        case mempoolEvent of
-          TraceMempoolRemoveTxs txs _ ->
-            -- TraceMempoolRemoveTxs are previously valid transactions that are no longer valid because of
-            -- changes in the ledger state. These transactions are already removed from the mempool,
-            -- so we can treat them as completely processed.
-            modifyMVar_ txsProcessedCounter $ \counter -> return $ counter + fromIntegral (length txs)
-          _ -> return ()
+
+    notifyTxsProcessed :: Tracer IO (TraceEventMempool blk)
+    notifyTxsProcessed = Tracer $ \case
+      TraceMempoolRemoveTxs [] _ -> return ()
+      TraceMempoolRemoveTxs txs _ -> do
+        -- TraceMempoolRemoveTxs are previously valid transactions that are no longer valid because of
+        -- changes in the ledger state. These transactions are already removed from the mempool,
+        -- so we can treat them as completely processed.
+        newCounter <- atomicModifyIORef' bcCounters $ \counters ->
+          let nc = bcTxsProcessedNum counters + fromIntegral (length txs)
+          in (counters { bcTxsProcessedNum = nc }, nc)
+        meta <- mkLOMeta Notice Public
+        traceNamedObject (appendName "metrics" tracer)
+                         (meta, LogValue "txsProcessedNum" (PureI $ fromIntegral newCounter))
```

Comment:
This piece of code is repeated in many places. Best to make a function out of it.